src/client/auth.js

```diff
@@ -1,16 +1,67 @@
 const UUID = require('uuid-1345')
 const yggdrasil = require('yggdrasil')
+const fs = require('mz/fs')
```

Comment:
The node `fs` api have a promisified version https://nodejs.org/api/fs.html#fs_fs_promises_api
You can use it with `require('fs/promises')` or ` require('fs').promises`

Reply:
Ah okay, this was just mirrored from the minecraft wrapper module, which also used mz